src/mudlet.pro

```diff
@@ -626,6 +636,24 @@ linux|macx|win32 {
     }
 }
 
+linux|macx|win32 {
+    contains( DEFINES, INCLUDE_3DMAPPER ) {
+        HEADERS += glwidget.h
+        SOURCES += glwidget.cpp
+        !build_pass{
+            message("The 3D mapper code is included in this configuration")
+        }
+    } else {
+        !build_pass{
+            message("The 3D mapper code is excluded from this configuration")
+        }
+    }
+} else {
+    !build_pass{
+        message("The 3D mapper code is excluded as OpenGL 1.5 might not be available on this platform")
```

Comment:
The bottom line is whether the platform supports OpenGL code that does **NOT** use the [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) a.k.a. OpenGL Shading Language...

Reply:
So what does it use? Need it for understandable wording here.

Reply:
By Open GL 4.0 (but I am not sure of the exact version number) implementations were allowed to drop support for *direct* OpenGL which does not use GLSL - and they (and all OpenGLES versions) only needed to support the *indirect* form that does use it. Somewhere between 2.0 and 4.0 it was expected to have a compatibility feature to handle the *old stuff*.

The awkwardness in porting from direct to indirect is that the indirect form needs the user to provide code (the GLSL) to set up the projection and viewport and other stuff that previously was done *simply* (?) with calls to OpenGL functions like `gluLookAt(...)`, `glViewport(...)`, `gluPerspective(...)` - and you really have to know what you are doing in order to get anything working.

Reply:
I'm still not sure what to better say here, how about "The 3D mapper code is excluded as this platform might be limited to GLSL only" ?

Reply:
Fair enough -  not perfect - but I guess it can do until we can modernise the OpenGL code. On that point I've just found http://www.mbsoftworks.sk/tutorials/opengl3/ which is a *mature* tutorial (aimed at Windows) but using this indirect stuff in OpenGL 3.x and the GLEW library (which is available at least on Debian) - so it ought to be portable - so I might have a look at that in the next few weeks to seek inspiration.